src/clj/puppetlabs/puppetserver/jruby_request.clj

```diff
@@ -45,6 +47,29 @@
      jruby-puppet jruby-service {:request (dissoc request :ssl-client-cert)}
      (handler (assoc request :jruby-instance jruby-puppet)))))
 
+(schema/defn ^:always-validate
+  wrap-with-concurrent-request-limit :- IFn
+  "Middleware fn that returns a 503 'Service Unavailable' response if the
+  number of outstanding requests for a JRuby instance exceed the given limit.
+  If splay-limit is set, the response will include a Retry-After header set
+  to a random fraction of the limit."
+  [handler :- IFn
+   metrics-service :- (schema/protocol jruby-metrics/JRubyMetricsService)
+   request-limit :- schema/Int
+   splay-limit :- schema/Int]
+  ;; TODO: Log warnings when limit is hit. Add i18n to user-facing strings.
+  (fn [request]
+    (let [metrics-map (jruby-metrics/get-metrics metrics-service)
+          requested-count (-> metrics-map :requested-instances deref count)]
```

Comment:
Do you think the `max-active-requests` limit should be checked against the sum of `requested-instances` and `borrowed-instances`, the latter of which could range up to whatever value is configured for `max-active-instances` in the pool, or just the `requested-instances`?  I think I'm good either way, just curious if you had thought about it.